### ISSUE TITLE
ValueError: unknown locale: en_IL on Ubuntu 18.04.1 LTS

### ISSUE THREAD

**Issue by Omer80** [spyder-ide/spyder#7602]
I can not open spyder in anaconda at ubuntu 18. I tried the solutions in [#4131](https://github.com/spyder-ide/spyder/issues/4131) and [#7051](https://github.com/spyder-ide/spyder/issues/7051), but it didn't work.

### What steps reproduce the problem?

1. I installed Anaconda for Python 2.7 on Ubuntu 18.04
2. Launch spyder
3. I get the following ValueError message regarding `unknown locale: en_IL`

```python-traceback
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
XmbTextListToTextProperty result code -2
Traceback (most recent call last):
  File "/home/ohm/anaconda2/lib/python2.7/site-packages/spyder/app/mainwindow.py", line 3208, in main
    mainwindow = run_spyder(app, options, args)
  File "/home/ohm/anaconda2/lib/python2.7/site-packages/spyder/app/mainwindow.py", line 3084, in run_spyder
    main.setup()
  File "/home/ohm/anaconda2/lib/python2.7/site-packages/spyder/app/mainwindow.py", line 1172, in setup
    self.setup_layout(default=False)
  File "/home/ohm/anaconda2/lib/python2.7/site-packages/spyder/app/mainwindow.py", line 1457, in setup_layout
    self.setup_default_layouts('default', settings)
  File "/home/ohm/anaconda2/lib/python2.7/site-packages/spyder/app/mainwindow.py", line 1633, in setup_default_layouts
    widget.toggle_view(True)
  File "/home/ohm/anaconda2/lib/python2.7/site-packages/spyder/plugins/ipythonconsole.py", line 725, in toggle_view
    self.create_new_client(give_focus=False)
  File "/home/ohm/anaconda2/lib/python2.7/site-packages/spyder/plugins/ipythonconsole.py", line 1046, in create_new_client
    self.connect_client_to_kernel(client, is_cython=is_cython)
  File "/home/ohm/anaconda2/lib/python2.7/site-packages/spyder/plugins/ipythonconsole.py", line 1074, in connect_client_to_kernel
    is_cython=is_cython)
  File "/home/ohm/anaconda2/lib/python2.7/site-packages/spyder/plugins/ipythonconsole.py", line 1509, in create_kernel_manager_and_kernel_client
    config=None, autorestart=True)
  File "/home/ohm/anaconda2/lib/python2.7/site-packages/traitlets/traitlets.py", line 958, in __new__
    inst.setup_instance(*args, **kwargs)
  File "/home/ohm/anaconda2/lib/python2.7/site-packages/traitlets/traitlets.py", line 986, in setup_instance
    super(HasTraits, self).setup_instance(*args, **kwargs)
  File "/home/ohm/anaconda2/lib/python2.7/site-packages/traitlets/traitlets.py", line 977, in setup_instance
    value.instance_init(self)
  File "/home/ohm/anaconda2/lib/python2.7/site-packages/traitlets/traitlets.py", line 1691, in instance_init
    self._resolve_classes()
  File "/home/ohm/anaconda2/lib/python2.7/site-packages/traitlets/traitlets.py", line 1696, in _resolve_classes
    self.klass = self._resolve_string(self.klass)
  File "/home/ohm/anaconda2/lib/python2.7/site-packages/traitlets/traitlets.py", line 1507, in _resolve_string
    return import_item(string)
  File "/home/ohm/anaconda2/lib/python2.7/site-packages/traitlets/utils/importstring.py", line 34, in import_item
    module = __import__(package, fromlist=[obj])
  File "/home/ohm/anaconda2/lib/python2.7/site-packages/jupyter_client/session.py", line 61, in <module>
    from jupyter_client.jsonutil import extract_dates, squash_dates, date_default
  File "/home/ohm/anaconda2/lib/python2.7/site-packages/jupyter_client/jsonutil.py", line 28, in <module>
    datetime.strptime("1", "%d")
  File "/home/ohm/anaconda2/lib/python2.7/_strptime.py", line 274, in <module>
    _TimeRE_cache = TimeRE()
  File "/home/ohm/anaconda2/lib/python2.7/_strptime.py", line 192, in __init__
    self.locale_time = LocaleTime()
  File "/home/ohm/anaconda2/lib/python2.7/_strptime.py", line 70, in __init__
    self.lang = _getlang()
  File "/home/ohm/anaconda2/lib/python2.7/_strptime.py", line 29, in _getlang
    return locale.getlocale(locale.LC_TIME)
  File "/home/ohm/anaconda2/lib/python2.7/locale.py", line 564, in getlocale
    return _parse_localename(localename)
  File "/home/ohm/anaconda2/lib/python2.7/locale.py", line 477, in _parse_localename
    raise ValueError, 'unknown locale: %s' % localename
ValueError: unknown locale: en_IL

```

## Versions

* Spyder version: 3.2.8
* Python version: 2.7.15
* Qt version: 5.9.5
* PyQt version: 5.9.2
* Operating System name/version: Ubuntu 18.04.1 LTS

**Comment by ccordoba12** [comment]
This is fixed in our latest version (3.3.0). Please update by opening a terminal and running there

    conda update spyder

**Comment by dinarior** [comment]
Experienced this with Spyder 3.3.1 , and python 2.7.15 . Ubuntu 18.04.
Workaround is to change to system to en_US.
